Cubical/Data/FinWeak/Properties.agda

```diff
@@ -0,0 +1,58 @@
+{-# OPTIONS --safe #-}
+module Cubical.Data.FinWeak.Properties where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Function
+
+open import Cubical.Data.Nat using (ℕ; zero; suc)
+open import Cubical.Data.FinWeak.Base
+import Cubical.Data.FinData as FD
+
+open import Cubical.Relation.Nullary
+
+private variable
+  n : ℕ
+
+FinPureIsProp : isProp (FinPure n)
+FinPureIsProp zero zero = refl
+FinPureIsProp (suc p) (suc q) = cong suc (FinPureIsProp p q)
+
+FinPure→FinData : FinPure n → FD.Fin n
+FinPure→FinData zero    = FD.zero
+FinPure→FinData (suc p) = FD.suc (FinPure→FinData p)
+
+FinWeak→FinData : Fin n → FD.Fin n
+FinWeak→FinData (pure p)   = FinPure→FinData p
+FinWeak→FinData (weaken p) = FD.weakenFin (FinWeak→FinData p)
+
+FinData→FinWeak : FD.Fin n → Fin n
+FinData→FinWeak {one} FD.zero = pure zero
+FinData→FinWeak {suc (suc n)} FD.zero = weaken (FinData→FinWeak FD.zero)
```

Comment:
You might want to write ```{suc (suc _)}``` here, to make it clear that the variable is not used and avoid confusion with the n from the signature of the function.